samples/subsys/senss/src/main.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <senss/senss.h>
+#include <stdio.h>
+
+static int data_event_callback(
+		senss_sensor_handle_t handle,
+		void *buf, int size, void *param)
+{
+	struct senss_sensor_info info;
+	senss_get_sensor_info(handle, &info);
+
+	printf("Received sensor data from sensor: %s, type: %d, instance: %d\n",
+		info.name,
+		info.type,
+		info.instance
+	);
+
+	if (info.type == SENSS_SENSOR_TYPE_MOTION_ACCELEROMETER_3D) {
+		struct senss_sensor_value_3d_int32* value = buf;
+		printf("Received sensor data readings :%d\n",
+			value->header.reading_count);
+		for (int i = 0; i < value->header.reading_count; i++) {
+			printf("Accel sensor data : x: %d, y: %d, z: %d\n",
+				value->readings[i].x,
+				value->readings[i].y,
+				value->readings[i].z);
+		}
+	}
+}
+
+void main(void)
+{
+	int ret;
+	ret = senss_init();
+	if (!ret) {
+		printf("Failed to init sensor subsystem! \n");
+		return;
+	}
+
+	/* Enumerate all sensor instances */
+	int sensor_count;
+	sensor_count = senss_get_sensor_count(SENSS_SENSOR_TYPE_ALL, -1);
```

Comment:
Could this be known at compile time to avoid needing any dynamic allocations below? Are the sensor counts actually dynamic?

Reply:
All sensors are configured at compile time.  There is no dynamic allocations for sensors in `senss` .

But some runtime information query APIs such as `senss_find_sensor`, `senss_get_sensor_count` are still very helpful,  for example Apps can use these APIs to dynamic enumerate sensors at runtime without code changes when sensor configuration changes.  

A real example is `CHRE` sensor `PAL` which need implement the `getSesnors` API: 
https://github.com/zephyrproject-rtos/chre/blob/zephyr/pal/include/chre/pal/sensor.h#L206